File: pythonStudy/Solved/BOJ/1011 Fly me to the Alpha Centauri.py
# 1011 Fly me to the Alpha Centauri
# https://www.acmicpc.net/problem/1011
from math import floor, sqrt
input = __import__('sys').stdin.readline
[print(floor(2*sqrt((lambda x, y: y-x)(*[int(i) for i in input().split()]))-1e-9)) for _ in range(int(input()))]

File: C++/Solved/BOJ/1011 Fly me to the Alpha Centauri.cpp
#include <bits/stdc++.h>
// Fly me to the Alpha Centauri
// https://www.acmicpc.net/problem/1011
using namespace std;
typedef long long ll;

// 다음 공간이동으로 이동할 수 있는 최대 거리의 역
// https://oeis.org/A033638
ll solve(ll k) {
    return floor(2 * sqrt(k) - 1e-9);
}
int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);

    int tc;
    ll a, b;
    cin >> tc;
    for (int i = 0; i < tc; i++) {
        cin >> a >> b;
        ll dist = b - a;
        cout << solve(dist) << '\n';
    }
}